### DISCUSSION TITLE
off-by-1 naming problem with Video Download Helper

### DISCUSSION THREAD

**Discussion by RickLangston1955** [aclap-dev/video-downloadhelper#1553] [Help and Q&A]
### Your question

There appears to be an off-by-1 file naming problem with VDH. 

Here is an example link, for a 1969 album from the Keef Hartley Band:

https://www.youtube.com/watch?v=Sb-umMBViX8

When I bring it up on Youtube, I see the list of separate tracks on the right-hand side, and the first track is called "Sacked". The second track is called "Born to Die". If I click on the Born to Die link on the right-hand side and start playing it, and then press pause, then I click on the VDH icon in the Firefox toolbar, it will only give me the choice of Sacked to be downloaded, without Born to Die being seen. If I click on the Download button anyway, a download will take place, and it will be named "Sacked - Youtube.mp4". However, it is actually Born to Die. This requires me to rename the file to avoid confusion. 

I've been seeing this problem for a while, and I had hoped it would have been fixed by now, but perhaps it's not been seen by others to have been reported.

### Addon & CoApp version and details

I am running Firefox 133.0.3 64-bit on Windows 10. 
I am running Video Download Helper 9.1.0.44  (licensed version).
Companion App v2.0.19

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

There are several other similar reports on here.  You need to learn to use the search facility on this forum.

As a circumvention -- a workaround, not a solution, just something that will avoid the problem -- you should use the VDH `Download As` function instead of the `Download` function.  That way, you'll get the chance to name the files correctly from the outset.  There's more discussion related to this at #665 & #814.  You would do well to study those discussions.

**Comment by RickLangston1955** [comment]
Thank you for your reply. However, the discussions you cited regard the target directory, not specifically how the downloaded file is being named. The problem I'm citing is specific to the Youtube "Play All" scenario where VDH is getting confused about the naming of the downloaded file. And although it is true that "Download As..." can be chosen from the dropdown and then the filename can be changed, this should not be a necessary action. It seems to be that there's a basic off-by-1 naming problem. Hopefully one of the developers will be able to replicate the scenario and decide whether they think there's a bug.

**Comment by Mattmjs** [comment]
It's a known problem that's been around a few months see #1132 , it does not appear to be a problem in the beta extension as I
noted in that discussion. If you want to stick with the stable extension then see this comment https://github.com/aclap-dev/video-downloadhelper/discussions/1132#discussioncomment-11404540
